gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ConfigBasedDatasetsFinder.java

```diff
@@ -298,13 +304,32 @@ protected void executeItertorExecutor(Iterator<Callable<Void>> callableIterator)
 
   /**
    * Helper funcition for converting datasetURN into URI
+   * Enable Pattern-like support.
    */
-  private URI datasetURNtoURI(String datasetURN) {
+  public List<URI> datasetURNtoURI(String datasetURN) {
+    Path mergedPath = PathUtils.mergePaths(this.commonRoot, new Path(datasetURN));
+    List<URI> resultURIs = new ArrayList<>();
     try {
-      return new URI(PathUtils.mergePaths(new Path(this.storeRoot), new Path(datasetURN)).toString());
+      // mergedPath is possibly to have pattern-format characters like '*'
+      FileStatus[] matchedPaths = fs.globStatus(mergedPath);
+      for (FileStatus matchedPath: matchedPaths) {
+        List<FileStatus> fileStatuses = FileListUtils.listFilesRecursively(fs, matchedPath.getPath(), new PathFilter() {
+          @Override
+          public boolean accept(Path path) {
+            return true;
+          }
+        });
+        for (FileStatus fileStatus : fileStatuses) {
+          resultURIs.add(new URI(PathUtils.getPathWithoutSchemeAndAuthority(fileStatus.getPath()).toString()));
```

Comment:
Please check whether stripping the scheme and authority is the proper thing to do here.